### DISCUSSION TITLE
Laser GRBL 6.2 bug?

### DISCUSSION THREAD

**Discussion by Bluerdg** [arkypita/LaserGRBL#2220] [General]
I'm running Laser Grbl on an older laptop I've set up as a dedicated control for my laser.  It has 6GB memory and runs windows 10 with no issues.  I installed current stable version of GRBL in Jan. and successfully ran a handful of projects and test runs for a new laser on that version.  Earlier this week I was prompted to install 6.1 when opening GRBL and yesterday I was prompted to install 6.2.  I upgraded each time.  After install of 6.2 I ran a simple project. I made a single pass burn to test my settings and program completely froze after completion.  I rebooted and performed a second single pass burn to complete the project and program froze again.   All is normal thru completion chime, yet program is completely unresponsive, Cannot reset, exit, control head, absolutlely nothing.  Based on the almost immediate upgrade from 6.1 to 6.2, there's obviously some bugs but I'm not seeing any discussion and online search has provided nothing.  Interested it anyone is having similar issues or has direction to a fix I'm not finding.

**Comment by arkypita** [comment]
HI I'm not having this problem and I see that several thousand people have already updated to the latest version, and if there was a big problem I would have already been informed. I have to say though that based on some feedback I'm receiving, and some of my personal feelings, there seems to have been a change in how the operating system manages LaserGRBL threads. Perhaps this is specific only to certain versions of Windows or to PCs with a certain update. The strange thing is that I always do minimal updates to LaserGRBL and try to touch key features as little as possible. I have a doubt about a small thing. I'll try to make a new version specifically to allow you a test, if you can get me your feedback we'll understand if it solve your issue. Give me a moment while I prepare the update.

**Reply by Bluerdg** [reply]
Wow, thanks.  I'll test and definitely let you know once I run a test.  I'm new to Laser GRBL (January),  Is 5.4 the latest stable version prior to 6.1?  That is the latest version I can find for reinstall.  I was checking path layout when I installed 6.1 and didn't run the laser so I don't know if it had the same issue.  Windows has been updating like mad the last week or so, I suppose it could be an issue with this machine and a recent patch but version I had before the 6.1 update worked great.

**Reply by LDHTHOPI** [reply]
I am having a similar issue.  My projects are not completing now.   It has been about three weeks since the project was able to finish.  I think it might have been the update? 

The project is halted with a pop-up window saying there is a problem with my board due to "Machine Alarm" and the words: "Abort During Cycle" in red and "sys.aborts" in purple in the command window (on the right).
![Screenshot (37)](https://github.com/arkypita/LaserGRBL/assets/163628131/92265bf7-aff4-4aa0-8e4d-c2365986ed53)

Sometimes the project will stop after 70% completion, sometimes after 50%, sometimes after only a few minutes.  It does not stop at the same time.  
 
I found the 5.4 version in the link you shared, but I do not know which file I need to download; nothing says "download."  I also tried to adjust the configuration, but that does not work.  

My machine does have the hard-stop switches on the X and Y, but the homing function will not work.  When I activate homing and click the button, the motor locks and makes a "hissing" noise?

Any help is greatly appreciated.  I am not very tech savvy and I do not understand code.

**Reply by arkypita** [reply]
It is a machine problem, and it has nothing to do with LaserGRBL (no need to download other versions, will be the same).
I don't know what the cause could be, you have to contact the manufacturer because it is a machine issue.

**Reply by LDHTHOPI** [reply]
Oh, shoot.   I bought the machine on Amazon and the product is no longer available.  I cannot contact the manufacturer.  Do I need a new main board?  Are you the creator of LaserGRBL?  Do you offer consulting services?

**Reply by LDHTHOPI** [reply]
These two red lights turn on when I plug the board into the laptop using a USB cable.  They turn on even when the power cord is not attached.  Should both lights be on?
![20240316_090259 1](https://github.com/arkypita/LaserGRBL/assets/163628131/f1e7ea2d-a3dd-40f5-89cc-bd18de51479b)

**Reply by LDHTHOPI** [reply]
I returned the engraver, I will purchase a new one.  How do I fully clear the LaserGRBL configuration settings?  I want the settings to be default when the new laser comes in.  The last time I uninstalled and reinstalled the software, the configuration settings were the same as before the uninstall.  Thanks.

**Comment by Bluerdg** [comment]
Sorry to bother you with this.  My laptop completed another windows update when I fired it up and just for giggles I ran a test on a previous project and it worked perfect. Just to be sure it wasn't something with the project file I ran a test on it as well and it worked perfect.  I will say that yesterday both times I ran a burn the laser jumped from 0/0 home point to upper right of a 4 corner burn and worked downward and back to 0/0.  The last test I ran started in the LOWER right and proceeded upward and finishing where it had started yesterday.   I truly appreciate your response.  I guess it has something to do with windows and their never ending updates and patches

**Reply by arkypita** [reply]
unfortunately I wasn't able to prepare the new version, and I won't be able to do so until Sunday afternoon. if you want in the meantime you can try installing a previous version and do some tests, and then install the next one to understand where the problem has arisen. the archive with all versions can be found here: https://github.com/arkypita/LaserGRBL/releases

 in my opinion we have the problem between version 5.7 and 6.0 

Let me know how it goes

**Comment by arkypita** [comment]
> Sorry to bother you with this. My laptop completed another windows update when I fired it up and just for giggles I ran a test on a previous project and it worked perfect. Just to be sure it wasn't something with the project file I ran a test on it as well and it worked perfect. I will say that yesterday both times I ran a burn the laser jumped from 0/0 home point to upper right of a 4 corner burn and worked downward and back to 0/0. The last test I ran started in the LOWER right and proceeded upward and finishing where it had started yesterday. I truly appreciate your response. I guess it has something to do with windows and their never ending updates and patches

Ok, thank you for your quick feedback.

Just to be sure I am going to release a minor version to undo some small changes (nothing that impact in how lasergrbl works) but that revert some settings that can impact on what libraries the OS should load for LaserGRBL.

**Comment by Bluerdg** [comment]
I had a similar issue on first run after the 6.2 update,  it completed
after a cold restart of my laptop but upon completion of a project the
program would freeze completely unresponsive.  Windows dropped an update as
I was trying to figure it out.  I completed that and performed a reinstall
of 6.2 and it works like a charm.

**Reply by LDHTHOPI** [reply]
I uninstalled the software and reinstalled, that did not fix the problem.  I did notice the configuration settings were the same after I reinstalled the software.   I thought an uninstall and reinstall would reset the settings to default.  Why would the settings be same after the reinstall?
